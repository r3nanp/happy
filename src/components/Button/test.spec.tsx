import { render, screen } from 'utils/tests/test-utils'
import { Button } from '.'

describe('<Button />', () => {
  it('Should be able to render a button', () => {
    const { container } = render(<Button>Button</Button>)

    expect(screen.getByRole('button', { name: /button/i })).toHaveStyle({
      height: '4rem',
      'border-radius': '1.5rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should render a button in the success page', () => {
    render(<Button isSuccessPage={true}>Button</Button>)

    expect(screen.getByText('Button')).toHaveStyleRule(
      'background-color',
      '#31b272'
    )
  })

  it('Should render a disabled Button', () => {
    render(<Button disabled>Button</Button>)

    expect(screen.getByRole('button', { name: /button/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled'
      }
    )
  })
})
