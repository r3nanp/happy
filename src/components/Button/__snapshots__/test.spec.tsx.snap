// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should be able to render a button 1`] = `
.c0 {
  margin-top: 4rem;
  width: 100%;
  height: 4rem;
  border: 0;
  background: #3CDC8C;
  border-radius: 1.5rem;
  padding: 0.8rem 3.2rem;
  color: #fff;
  font-weight: 800;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
}

.c0:not(:disabled):hover {
  background: #36cf82;
}

<button
  class="c0"
>
  Button
</button>
`;
